sdk/keyvault/keyvault-common/src/challengeBasedAuthenticationPolicy.ts

```diff
@@ -50,8 +50,12 @@ type ChallengeState =
  *
  * Following the first request of a client, follow-up requests will get the cached token
  * if possible.
+ *
+ * @param verifyChallengeResource - whether to verify the challenge resource matches the Key Vault or Managed HSM domain.
  */
-export function createChallengeCallbacks(): ChallengeCallbacks {
+export function createChallengeCallbacks(
+  verifyChallengeResource: boolean = true
```

Comment:
It may be worth putting `verifyChallengeResource` into an options bag. That way if more options are added in the future we could do it easier